docs: set explicit ssl_policy in network load balancer example (#38806)
hashicorp/terraform-provider-aws: website/docs/r/lb_listener.html.markdown

* `mutual_authentication` - (Optional) The mutual authentication configuration information. Detailed below.
* `port` - (Optional) Port on which the load balancer is listening. Not valid for Gateway Load Balancers.
* `protocol` - (Optional) Protocol for connections from clients to the load balancer. For Application Load Balancers, valid values are `HTTP` and `HTTPS`, with a default of `HTTP`. For Network Load Balancers, valid values are `TCP`, `TLS`, `UDP`, and `TCP_UDP`. Not valid to use `UDP` or `TCP_UDP` if dual-stack mode is enabled. Not valid for Gateway Load Balancers.

> * `ssl_policy` - (Optional) Name of the SSL Policy for the listener. Used in combination with `HTTPS` or `TLS` value for `protocol`. Default is `ELBSecurityPolicy-2016-08`.

Reviewer: Specifying the default value is great, developers will not like to scour the web again to find the default value. Hence it improves the adoption of this project. However, I believe the last sentence is all that is required here. The first part of the sentence you suggested is confusing. The previous sentence is clear and direct.
```suggestion
* `ssl_policy` - (Optional) Name of the SSL Policy for the listener. Required if `protocol` is `HTTPS` or `TLS`. Default is `ELBSecurityPolicy-2016-08`.
```

Author: Hi @network-charles,

thanks for your feedback!

- I agree on the default value: when users do not specify the `ssl_policy`, they should at least know what value will be used then.
- On the "Required if `protocol` is `HTTPS` or `TLS`" change:  If you think its more confusing, I can remove that change. I was thinking that listing the argument as "Optional" and then explaining its "Required" in some cases is not ideal.

Reviewer: I understand, but it's correct because In certain situations, it is required to have it, but in general, it is optional.

Author: I changed back to the original version.

In which situations would one need to specify the `ssl_policy` explicitly/ when is it required? 
For the NLB/ TLS example I provided a code snippet showing that it not required.

Reviewer: When the developer needs to use a TLS 1.3 for example.
